addons/mail/static/src/core/common/im_status_service_patch.js

```diff
@@ -7,30 +7,30 @@ export const imStatusServicePatch = {
     dependencies: [...imStatusService.dependencies, "mail.store"],
 
     start(env, services) {
-        const { bus_service, "mail.store": store, presence } = services;
         const API = super.start(env, services);
-
-        bus_service.subscribe(
+        services["bus_service"].subscribe(
             "bus.bus/im_status_updated",
-            ({ im_status, partner_id, guest_id }) => {
-                const persona = store.Persona.get({
-                    type: partner_id ? "partner" : "guest",
-                    id: partner_id ?? guest_id,
-                });
-                if (!persona) {
-                    return; // Do not store unknown persona's status
-                }
-                persona.im_status = im_status;
```

Comment:
Why not just move this LOC into a new method and overwrite that. It saves us from duplicating the code and less maintainance.